unified_planning/engines/factory.py

```diff
@@ -362,13 +364,14 @@ def _add_meta_engine(
 
     def _get_engine_class(
         self,
-        engine_kind: str,
+        engine_kind: "OperationMode",
         name: Optional[str] = None,
         problem_kind: ProblemKind = ProblemKind(),
         optimality_guarantee: Optional["OptimalityGuarantee"] = None,
         compilation_kind: Optional["CompilationKind"] = None,
         plan_kind: Optional["PlanKind"] = None,
         anytime_guarantee: Optional["AnytimeGuarantee"] = None,
+        operation_mode: Optional["OperationMode"] = None,
```

Comment:
I would rename this, for the inetrnal methods, while for the public ones it's OK: 
```suggestion
        target_operation_mode: Optional["OperationMode"] = None,
```